hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/table/ITTestHoodieDataSource.java

```diff
@@ -1129,9 +1113,9 @@ void testWriteLocalIndex() {
     assertRowsEquals(result, expected, 3);
   }
 
-  @Test
-  void testStreamReadEmptyTablePath() throws Exception {
-    // case1: table metadata path does not exists
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  void testStreamReadEmptyTablePath(boolean initTable) throws Exception {
```

Comment:
what the purpose for this test change?

Reply:
For the previous code, sometimes tm resources are not released after case1 canceled, which lead to unexpected failure of case2. The change here is to make sure a new table environment is created for testing every case, and enhance the stability.